docs/management_api.yml

```diff
@@ -527,6 +527,74 @@ paths:
         500:
           $ref: "#/responses/InternalServerError"
 
+  /artifacts/generate:
+    post:
+      summary: Upload raw data to generate a new artifact
+      description: |
+        Generate a new Mender artifact from raw data and meta data. Multipart request with meta and raw file.
+        Supports generating single-file updates only, using the single file upload module. [https://hub.mender.io/t/single-file/486]: 
+      consumes:
+        - multipart/form-data
+      parameters:
+        - name: Authorization
+          in: header
+          required: true
+          type: string
+          format: Bearer [token]
+          description: Contains the JWT token issued by the User Administration and Authentication Service.
+        - name: name
+          in: formData
+          description: Name of the artifact to generate.
+          required: true
+          type: string
+        - name: description
+          in: formData
+          description: Description of the artifact to generate.
+          required: false
+          type: string
+        - name: size
```

Comment:
afaik this field and its handling is obsolete
we have this field in the endpoint for uploading artifacts just because at the time when we're implementing artifact upload golang was not providing info about the size of the file in form data, but that has changed some time ago and now the info about the size is available

Reply:
@kjaskiewiczz I wasn't aware of this historical detail, it is a required field for `POST /artifacts` as well as in the internal api `POST /tenants/{id}/artifacts` so I used it for the new end-point as well. What shall we do, in your opinion? Shall we remove it, or make it non-required? Just for this end-point or for all of them?

Reply:
I think we can just drop it in this endpoint.

since we have some shared implementation underneath - if that poses a problem, we can also make it optional in other endpoints. if I understand correctly, it's a non breaking change, so it's ok to modify `v1` in this way and still keep it `v1`.

but if it's straightforward to just remove it from here, I'd do that.

Reply:
@mchalski ack, creating a ticket in JIRA to keep track of this.

Reply:
ideally, we could remove it from here and make it optional in existing endpoints

Reply:
Created MEN-3036 and added to the current sprint.